### PR TITLE
Fix machine creation bootstrap tokens expiration

### LINKED ISSUE #772: Node cannot join the cluster due to expired bootstrap token
**How to categorize this issue?**

/kind bug
/priority 3

**What happened**:
Mostly always my machines take more than 20 minutes to be created in the cloudprovider.
Therefore the workers are configured to with a specific `machineCreationTimeout` of `50m`.
When the nodes come up after approx. 30min the node is unable to join cluster due to a expired bootstrap token.

**What you expected to happen**:

Bootstrap token for node does not expired before the configured `spec.worker[].machineController.machineCreationTimeout`

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a shoot and configure a worker group with a `machineCreationTimeout` of more than 20m
2. let the machine need more than 20 minutes to be created

**Anything else we need to know?**:

**Environment**:

- Kubernetes version (use `kubectl version`): 1.22.17
- Cloud provider or hardware configuration: Equinix

### PR DESCRIPTION
**Pull request by schrodit**
**What this PR does / why we need it**:

The expiration of bootstrap tokens is hardcoded in the machine controller deployment. (depending on the provider but mostly approx. 20min).
There is an option to overwrite this default per worker group with `machineCreationTimeout`.
This option is considered during the machine creation but not in the bootstrap token.
So this means that if the machine takes more time than 20 minutes to be created, the bootstrap token is already expired and the needed configs cannot be fetched form the cluster.

This PR makes the bootstrap token respect the machines `machineCreationTimeout` option.

**Which issue(s) this PR fixes**:
Fixes #772

**Special notes for your reviewer**:

Tested with the equinix provider

**Release note**:

```bugfix user
Fix a bug in the bootstrap token creation that caused node to not be able to join the cluster due to an expired bootstrap token.
```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @schrodit for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: pkg/util/provider/machinecontroller/machine_bootstrap_token.go [hunk 1]
```diff
@@ -90,11 +91,15 @@ func (c *controller) getBootstrapTokenOrCreateIfNotExist(ctx context.Context, ma
 			if err != nil {
 				return nil, err
 			}
+			expiration := c.safetyOptions.MachineCreationTimeout.Duration
+			if machine.Spec.MachineConfiguration != nil && machine.Spec.MachineConfiguration.MachineCreationTimeout != nil {
+				expiration = machine.Spec.MachineConfiguration.MachineCreationTimeout.Duration
+			}
```
Comment (himanshu-kun) [inline pkg/util/provider/machinecontroller/machine_bootstrap_token.go:97]: we can use the helper function `getEffectiveCreationTimeout` for this.

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
It is a serious bug , Thanks for the PR Tim!

File: pkg/util/provider/machinecontroller/machine_bootstrap_token.go (hunk 1 above)
Comment (schrodit) [reply pkg/util/provider/machinecontroller/machine_bootstrap_token.go]: ah didn't know that helper function exists.

Updated the code PTAL.

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Pull request merged**